### DISCUSSION TITLE
Switching between front and back facing camera

### DISCUSSION THREAD

**Discussion by Kavilugu** [BlackBoxVision/react-qr-reader#7] [Ideas]
Hello! 

Just started using this component for a PWA prototype and I noticed that on my android (Pixel 2) the camera defaults to front facing, which is not ideal for scanning QR codes. 

Would there be a way to switch this or is it limited by the Zxing library?

Thanks in advance!

**Comment by JonatanSalas** [comment]
Hi, @Kavilugu thanks for using our library. The `QrReader` component has a prop that is called `facingMode` which you can set to `user` or `environment`. In the first case, it would take the front camera, and in the second case it would take the rear/back camera. 

Let me know if setting this prop does the work for you.

**Reply by JonatanSalas** [reply]
If this doesn't work you can try the next version of the library, whose current version is `4.0.0-2`. In that version we've made a breaking change that lets users pass its own [`MediaTrackConstraints`](https://developer.mozilla.org/en-US/docs/Web/API/MediaTrackConstraints) backed by a default value of `{ facingMode: 'user' }`.

**Reply by Kavilugu** [reply]
Thanks for the reply!

I've unfortunately already tried passing both ```environment``` as well as ```user``` to ```facingMode``` and they both display the front camera. 

I'll have to look into the next version, thanks!

**Reply by JonatanSalas** [reply]
You're welcome, if it isn't working please open an issue and we'll try to figure out the fix as soon as we can

**Reply by JonatanSalas** [reply]
I've made some minor changes and finally released the v4, you can install it by doing `npm i @blackbox-vision/react-qr-reader@4.0.0`

**Reply by Kavilugu** [reply]
Much appreciated! 
I'll try it as soon as I get home and let you know if this fixes the issue.

**Reply by JonatanSalas** [reply]
You're welcome! I hope this fixes your issue.

**Reply by Kavilugu** [reply]
I saw that an issue was opened and closed but just wanted to confirm that it fixed it for my use case as well!
Thanks for the quick fix!

**Reply by JonatanSalas** [reply]
You're welcome, glad to help you!
